### ISSUE TITLE
History variable `FATES_TRANSITION_MATRIX_LULU` contains garbage for `FatesColdDryDepSatPhen` and `FatesColdMeganSatPhen`

### ISSUE THREAD

**Issue by slevis-lmwg** [ESCOMP/CTSM#2656]
> Yep, those cases are definitely garbage, but they've been that way for these two tests since that variable was introduced in `ctsm5.2.013`. Those values should be zero'd out. I'm not sure why the `FatesColdSatPhen` correctly zeros `FATES_TRANSITION_MATRIX_LULU`, but `FatesColdDryDepSatPhen` and `FatesColdMeganSatPhen` does not.  

_Originally posted by @glemieux in https://github.com/ESCOMP/CTSM/issues/2605#issuecomment-2243842711_

**Comment by slevis-lmwg** [comment]
I originally spotted the problem in #2605 as a strange difference from baseline in these two tests:
```
SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdDryDepSatPhen.GC.0719-175606de_gnu/SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdDryDepSatPhen.GC.0719-175606de_gnu.clm2.h0.2000-01-01-00000.nc.cprnc.out: RMS FATES_TRANSITION_MATRIX_LULU       Infinity            NORMALIZED    Infinity
SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdMeganSatPhen.GC.0719-175606de_gnu/SMS_D.1x1_brazil.I2000Clm60FatesSpCruRsGs.derecho_gnu.clm-FatesColdMeganSatPhen.GC.0719-175606de_gnu.clm2.h0.2000-01-01-00000.nc.cprnc.out: RMS FATES_TRANSITION_MATRIX_LULU       Infinity            NORMALIZED    Infinity
```

**Comment by glemieux** [comment]
Checking the difference between `FatesColdSatPhen` testmod, which appropriately zeros `FATES_TRANSITION_MATRIX_LULU`,  and these failing testmods, the only difference are the expected settings in shell_commands (i.e. `CLM_BLDNML_OPTS` set to `'--drydep'` and `'--megan'`).

**Comment by glemieux** [comment]
This might simply be an initialization error on the fates side that is only showing up with `gnu` compiler tests.  I noticed that the instance of `FatesColdSatPhen` are either `intel` or `nvhpc`.  Testing.

**Comment by glemieux** [comment]
> This might simply be an initialization error on the fates side that is only showing up with `gnu` compiler tests. I noticed that the instances of `FatesColdSatPhen` that I've been checking are either `intel` or `nvhpc`. Testing.

It looks like this is actually the case.  Running `gnu` compiled version of the `FatesColdSatPhen` test results in the transition matrix reporting junk.  I'll open an issue on the fates-side and cross reference here.

**Comment by glemieux** [comment]
This probably is being missed in more tests that are based on the `Fates` testmod as history output is being cleared and this output is not added.

**Comment by glemieux** [comment]
fates issue: https://github.com/NGEET/fates/issues/1230

**Comment by glemieux** [comment]
Note that this will be resolved via https://github.com/NGEET/fates/pull/1231, which will likely be fates tag `sci.1.77.2_api.36.0.0`.  This tag could come in with #2594.

**Comment by glemieux** [comment]
This is fixed on fates main via `sci.1.77.2_api.36.0.0`.

**Comment by glemieux** [comment]
This should be fixed by #2624
